coderd/prometheusmetrics/prometheusmetrics_test.go

```diff
@@ -352,3 +353,85 @@ func TestAgents(t *testing.T) {
 		return agentsUp && agentsConnections && agentsApps && agentsExecutionInSeconds
 	}, testutil.WaitShort, testutil.IntervalFast)
 }
+
+func TestAgentStats(t *testing.T) {
+	t.Parallel()
+
+	// Build a sample workspace with test agent and fake agent client
+	client, _, api := coderdtest.NewWithAPI(t, &coderdtest.Options{IncludeProvisionerDaemon: true})
+	db := api.Database
+
+	user := coderdtest.CreateFirstUser(t, client)
+	authToken := uuid.NewString()
+	version := coderdtest.CreateTemplateVersion(t, client, user.OrganizationID, &echo.Responses{
+		Parse:          echo.ParseComplete,
+		ProvisionPlan:  echo.ProvisionComplete,
+		ProvisionApply: echo.ProvisionApplyWithAgent(authToken),
+	})
+	template := coderdtest.CreateTemplate(t, client, user.OrganizationID, version.ID)
+	coderdtest.AwaitTemplateVersionJob(t, client, version.ID)
+	workspace := coderdtest.CreateWorkspace(t, client, user.OrganizationID, template.ID)
+	coderdtest.AwaitWorkspaceBuildJob(t, client, workspace.LatestBuild.ID)
+
+	agentClient := agentsdk.New(client.URL)
+	agentClient.SetSessionToken(authToken)
+
+	registry := prometheus.NewRegistry()
+
+	// given
+	cancel, err := prometheusmetrics.AgentStats(context.Background(), slogtest.Make(t, nil), registry, db, time.Now(), time.Second)
+	require.NoError(t, err)
+	t.Cleanup(cancel)
+
+	// when
+	_, err = agentClient.PostStats(context.Background(), &agentsdk.Stats{
```

Comment:
I think the test should have two+ agents and multiple posted metrics for each to see how the queries can handle it.

Reply:
Done. Now we have 3 agents and some random numbers. There is also a golden file to match the results.